WcaOnRails/app/controllers/api/v0/users_controller.rb

```diff
@@ -29,37 +23,43 @@ def show_user_by_wca_id
   end
 
   def permissions
-    if current_user
-      if stale?(current_user)
-        render json: current_user.permissions
-      end
-    else
-      render status: :unauthorized, json: { error: I18n.t('api.login_message') }
+    require_user!
+    if stale?(current_user)
+      render json: current_user.permissions
     end
   end
 
+  def personal_records
+    require_user!
+    return render json: { single: [], average: [] } unless current_user.wca_id.present?
+    person = Person.includes(:user, :ranksSingle, :ranksAverage).find_by_wca_id!(current_user.wca_id)
+    render json: { single: person.ranksSingle, average: person.ranksAverage }
+  end
+
+  def preferred_events
+    require_user!
+    preferred_events = Rails.cache.fetch("#{current_user.id}-preferred", expires_in: 24.hours) do
```

Comment:
Rails cache key doesn't say what is being preferred. Preferred... Delegates? Preferred... home cooking recipes?